targets/RedundantADC/main.cpp

```diff
@@ -0,0 +1,46 @@
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/UART.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/manager.hpp>
+#include <dev/RedundantADC.hpp>
+
+namespace IO = EVT::core::IO;
+namespace DEV = EVT::core::DEV;
+
+int main() {
+    // Initialize system
+    EVT::core::platform::init();
+
+    // Setup UART
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    uart.printf("UART initialized\r\n");
+
+    IO::ADC& adc0 = IO::getADC<IO::Pin::PA_0>();
+    uart.printf("ADC0 initialized\r\n");
+    IO::ADC& adc1 = IO::getADC<IO::Pin::PA_1>();
+    uart.printf("ADC1 initialized\r\n");
+    IO::ADC& adc2 = IO::getADC<IO::Pin::PA_4>();
+    uart.printf("ADC2 initialized\r\n");
+
+    // Create RedundantADC object
+    HIB::DEV::RedundantADC redundantADC(adc0, adc1, adc2);
+
+    // Variables to store ADC values
+    uint32_t return_val;
+
+    while (1) {
+        // Process ADC values
+        HIB::DEV::RedundantADC::Status status = redundantADC.readVoltage(return_val);
+
+        //check ADC Statuses0
+        if (status == HIB::DEV::RedundantADC::Status::OK) {
```

Comment:
A switch-case here would be a bit cleaner, but it doesn't really matter

Reply:
if > Switch

Reply:
Objectively incorrect